src/uu/mv/src/mv.rs

```diff
@@ -98,7 +104,69 @@ pub enum OverwriteMode {
     ///'-f' '--force'         overwrite without prompt
     Force,
 }
+struct VerboseContext<'a> {
+    backup: Option<&'a Path>,
+    pb: Option<&'a MultiProgress>,
+}
+
+impl<'a> VerboseContext<'a> {
+    fn new(backup: Option<&'a Path>, pb: Option<&'a MultiProgress>) -> VerboseContext<'a> {
+        VerboseContext { backup, pb }
+    }
+
+    fn print_move_file(&self, from: &Path, to: &Path) {
+        let message = match self.backup.as_ref() {
+            Some(path) => format!(
+                "renamed {} -> {} (backup: {})",
+                from.quote(),
+                to.quote(),
+                path.quote()
+            ),
+            None => format!("renamed {} -> {}", from.quote(), to.quote()),
+        };
+
+        match self.pb {
+            Some(pb) => pb.suspend(|| {
+                println!("{message}");
+            }),
+            None => println!("{message}"),
+        };
+    }
+
+    fn print_copy_file(&self, from: &Path, to: &Path, with_backup_message: bool) {
+        let message = match self.backup.as_ref() {
+            Some(path) if with_backup_message => format!(
+                "copied {} -> {} (backup: {})",
+                from.quote(),
+                to.quote(),
+                path.quote()
+            ),
+            _ => format!("copied {} -> {}", from.quote(), to.quote()),
+        };
+
+        match self.pb {
+            Some(pb) => pb.suspend(|| {
+                println!("{message}");
+            }),
+            None => println!("{message}"),
+        };
+    }
+
+    fn create_folder(&self, path: &Path) {
```

Comment:
Directories are abbreviated as `dir` everywhere else. Is a folder something different, or a synonym?